src/gui/styles/types/custom_style.rs

```diff
@@ -0,0 +1,289 @@
+//! Custom Sniffnet color schemes.
+//! Themes should be in TOML files with the following schema.
+//! ```toml
+//! name = "Theme's name"
+//!
+//! [description]
+//! # Descriptions of the theme in any of Sniffnet's supported languages
+//! # CODE = "DESC"
+//! EN = "A fantastically dark theme!"
+//! HU = ""
+//! PL = ""
+//!
+//! # The palette is specified in RGB hexadecimal with an optional alpha
+//! [palette]
+//! primary = "#000000"
+//! secondary = "#000000"
+//! buttons = "#000000"
+//! outgoing = "#000000"
+//! text_headers = "#000000"
+//! text_body = "#000000"
+//! round_borders = "#000000"
+//! round_containers = "#000000"
+//! starred = "#000000"
+//!
+//! # floats for these two alpha values
+//! badge_alpha = 0.9
+//! color_mix_chart = 0.3
+//! ```
+
+use serde::{de::Error as DeErrorTrait, Deserialize, Deserializer, Serialize, Serializer};
+use std::{
+    collections::BTreeMap,
+    fs::{self, File},
+    io::{self, BufReader, Read},
+    path::PathBuf,
+};
+
+use super::palette::{Palette, PaletteExtension};
+use crate::Language;
+
+/// Custom color scheme data including the palette, name, and location of the toml.
```

Comment:
Sounds good! I have a question before I start. I implemented a `CustomStyle` struct that holds the name of the style, the translated descriptions, and the `Palette` plus the extra yellow color.

This struct made it easier to define themes without having to extend functions like `get_colors` for each theme.

**Original:**
```rust
pub fn get_colors(style: StyleType) -> Palette {
    match style {
        StyleType::Night => NIGHT_STYLE,
        StyleType::Day => DAY_STYLE,
        StyleType::DeepSea => DEEP_SEA_STYLE,
        StyleType::MonAmour => MON_AMOUR_STYLE,
    }
}
```

**How it looks in my PR:**
```rust
pub fn get_colors(style: &StyleType) -> &Palette {
    match style {
        StyleType::Night => &NIGHT_STYLE,
        StyleType::Day => &DAY_STYLE,
        StyleType::DeepSea => &DEEP_SEA_STYLE,
        StyleType::MonAmour => &MON_AMOUR_STYLE,
        StyleType::Custom(style) => &style.palette.base,
    }
}
```

Another example.
**Original:**
```rust
pub fn get_starred_color(style: StyleType) -> Color {
    match style {
        StyleType::Night | StyleType::DeepSea => Color {
            r: 245.0 / 255.0,
            g: 193.0 / 255.0,
            b: 39.0 / 255.0,
            a: 0.5,
        },
        StyleType::Day | StyleType::MonAmour => Color {
            r: 245.0 / 255.0,
            g: 193.0 / 255.0,
            b: 39.0 / 255.0,
            a: 0.8,
        },
    }
}
```

**How it looks in my PR:**
```rust
pub fn get_starred_color(style: &StyleType) -> Color {
    match style {
        StyleType::Night | StyleType::DeepSea => Color {
            r: 245.0 / 255.0,
            g: 193.0 / 255.0,
            b: 39.0 / 255.0,
            a: 0.5,
        },
        StyleType::Day | StyleType::MonAmour => Color {
            r: 245.0 / 255.0,
            g: 193.0 / 255.0,
            b: 39.0 / 255.0,
            a: 0.8,
        },
        StyleType::Custom(style) => style.palette.extension.starred,
    }
}
```

In other words, the struct encapsulates the palette plus the metadata for the theme. Functions take `&Arc<StyleType>` or `&StyleType` instead of `StyleType`.

Should I port that over to the new PR or just extend the `match`es with new themes as in the original code?

Reply:
The `Arc` should not be needed for the moment.

You can also forget about the palettes' descriptions since I'd like to include them just for the original themes, to not make complex getting all the translations from native speaker and because I'll display the additional palettes in a more condensed form.
The palette name is enough.

And we don't even need the path at this point.

I think the best thing to do is not keeping `CustomStyle`, but only keeping `CustomPalette`.

Reply:
Also, I know that's a lot limited with respect to all the work you did, but I think it's a good compromise given the simplicity of this solution and the fact that without a file dialog is not comfortable to include external palettes.

Reply:
I thought I responded to this. :sweat_smile:

It's no problem! A minimal, working custom styles implementation is best for now. Plus, I learned a lot anyway. I mostly finished the patch so I'll open a PR now.